packages/block-editor/src/components/inserter/menu.js

```diff
@@ -319,12 +323,13 @@ export class InserterMenu extends Component {
 				onKeyDown={ this.onKeyDown }
 			>
 				<div className="block-editor-inserter__main-area">
-					<label
-						htmlFor={ `block-editor-inserter__search-${ instanceId }` }
-						className="screen-reader-text"
-					>
-						{ __( 'Search for a block' ) }
-					</label>
+					<VisuallyHidden>
```

Comment:
It looks like this can be defined this as:
```jsx
<VisuallyHidden as="label" htmlFor={ `block-editor-inserter__search-${ instanceId }` }>
    { __( 'Search for a block' ) }
</VisuallyHidden>
```

which avoids the extra span around the label. I saw a few other places in the PR where that would be beneficial.

I noticed the `VisuallyHidden` docs don't mention this, could be a good follow up PR to update those docs 😄